modules/common/karpenter/main.tf

```diff
@@ -0,0 +1,171 @@
+locals {
+  node_subnet_ids = slice(var.subnet_ids, 0, var.az_count)
+  node_classes = {
+    cpu = {
+      name   = "cpu"
+      ami_id = data.aws_ami.default_eks_node.id
+    }
+    gpu = {
+      name   = "gpu"
+      ami_id = data.aws_ami.gpu_eks_node.id
+    }
+  }
+
+  cpu_instance_requirements = {
+    instance-family = {
+      key       = "karpenter.k8s.aws/instance-family"
+      operator  = "In"
+      values    = ["m5", "m6i", "m6a", "m7i", "m7a", "c5", "c6i", "c6a", "c7i", "c7a", "c7i-flex", "r5", "r6i", "r6a", "r7i", "r7a"]
```

Comment:
I'm still not sure that we should have so many potential node types. This is going to make it very difficult for @dzereski to know which reservations to make.

Reply:
We are now using saving plans according to Chris Wann https://indicodata.slack.com/archives/C022VV4Q6L8/p1740170496829399?thread_ts=1740164822.953139&cid=C022VV4Q6L8

So having a wider range of instances gives Karpenter more space to work with